# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/svg/SVGClipPathFrame.h
# Commit: 0fdbf0a455f2
# Full Hash: 0fdbf0a455f26afa0a7a0390d6a380e318fce9d4
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-10-24 03:54:44
# Regressor Bug: 1765202
# File Overlap Count: 1
# Description:
#   Bug 1765202 part 3 - ignore clip-path and mask properties that don't point to clipPath and mask elements r=emilio
#   
#   This refactors the logic for testing whether something has a clip-path or mask and what to do about it in one place rather than using many overlapping functions.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D190767
# ==============================================================================

diff -r 2a8d11fcc746 -r 0fdbf0a455f2 layout/svg/SVGClipPathFrame.h
--- a/layout/svg/SVGClipPathFrame.h	Mon Oct 23 20:32:20 2023 +0000
+++ b/layout/svg/SVGClipPathFrame.h	Mon Oct 23 20:32:37 2023 +0000
@@ -111,8 +111,6 @@
   // mask based clipping.
   bool IsTrivial(ISVGDisplayableFrame** aSingleChild = nullptr);
 
-  bool IsValid();
-
   // nsIFrame interface:
   nsresult AttributeChanged(int32_t aNameSpaceID, nsAtom* aAttribute,
                             int32_t aModType) override;
@@ -152,6 +150,8 @@
   void PaintChildren(gfxContext& aMaskContext, nsIFrame* aClippedFrame,
                      const gfxMatrix& aMatrix);
 
+  bool IsValid();
+
   // Set, during a GetClipMask() call, to the transform that still needs to be
   // concatenated to the transform of the DrawTarget that was passed to
   // GetClipMask in order to establish the coordinate space that the clipPath